src/python/grpcio_tests/tests_aio/unit/client_unary_stream_interceptor_test.py

```diff
@@ -0,0 +1,404 @@
+        assert call.done()
+        first_callback_ran.set()
+
+    second_callback_ran = asyncio.Event()
+
+    def second_callback(call):
+        # Validate that all resopnses have been received
+        # and the call is an end state.
+        assert call.done()
+        second_callback_ran.set()
+
+    call.add_done_callback(first_callback)
+    call.add_done_callback(second_callback)
+
+    async def validation():
+        await asyncio.wait_for(
+            asyncio.gather(first_callback_ran.wait(),
+                           second_callback_ran.wait()),
+            test_constants.SHORT_TIMEOUT)
+
+    return validation()
+
+
+class _UnaryStreamInterceptorEmpty(aio.UnaryStreamClientInterceptor):
+
+    async def intercept_unary_stream(self, continuation, client_call_details,
+                                     request):
+        return await continuation(client_call_details, request)
+
+
+class _UnaryStreamInterceptorWith_ResponseIterator(
+        aio.UnaryStreamClientInterceptor):
+
+    def __init__(self):
+        self.response_iterator = None
+
+    async def intercept_unary_stream(self, continuation, client_call_details,
+                                     request):
+        call = await continuation(client_call_details, request)
+        self.response_iterator = _ResponseIterator(call)
+        return self.response_iterator
+
+
+class TestUnaryStreamClientInterceptor(AioTestBase):
+
+    async def setUp(self):
+        self._server_target, self._server = await start_test_server()
+
+    async def tearDown(self):
+        await self._server.stop(None)
+
+    async def test_intercepts(self):
+        for interceptor_class in (_UnaryStreamInterceptorEmpty,
+                                  _UnaryStreamInterceptorWith_ResponseIterator):
+
+            with self.subTest(name=interceptor_class):
+                interceptor = interceptor_class()
+
+                request = messages_pb2.StreamingOutputCallRequest()
+                for _ in range(_NUM_STREAM_RESPONSES):
+                    request.response_parameters.append(
+                        messages_pb2.ResponseParameters(
+                            size=_RESPONSE_PAYLOAD_SIZE))
+
+                channel = aio.insecure_channel(self._server_target,
+                                               interceptors=[interceptor])
+                stub = test_pb2_grpc.TestServiceStub(channel)
+                call = stub.StreamingOutputCall(request)
+
+                await call.wait_for_connection()
+
+                response_cnt = 0
+                async for response in call:
+                    response_cnt += 1
+                    self.assertIs(type(response),
+                                  messages_pb2.StreamingOutputCallResponse)
+                    self.assertEqual(_RESPONSE_PAYLOAD_SIZE,
+                                     len(response.payload.body))
+
+                self.assertTrue(response_cnt, _NUM_STREAM_RESPONSES)
```

Comment:
Typo? `assertEqual`?

Reply:
now is when all of the tests fail :) I will fix it